docs/book/src/plugins/test-plugin.md

```diff
@@ -0,0 +1,130 @@
+# Write Tests For Your Own Plugins
+
+The Kubebuilder project tests plugins in multiple perspectives.
+
+First, the plugin's functionality is verified through E2E tests which will be performed in clusters of different K8s versions.
+
+Secondly, the scaffolded output from the plugin should be in consistent with multiple platforms.
```

Comment:
Hi @Kavinjsir see that KB can be used as a LIB.
Therefore, is possible to import KB in another repo and create your own plugin.
Then, after create your own plugin you will want to test that.
So, that we can share with the authors / its consumers that KB lib also provide the features to help out and then can do the tests such as / similarly we do in KB repo but in their own thing.

That is the goal of this doc. Is that make sense?

Reply:
However, we also could have a section like CONTRIBUTION where we could add the info on how to create new plugins on KB repo, how to test them out and etc.. that would be amazing actually.

Reply:
Yeah, that makes sense. Thx for clarifying!

Reply:
Hi @camilamacedo86 . I tried adding a section under [CONTRIBUTING.md](https://github.com/Kavinjsir/kubebuilder/blob/docs%2Ftest-plugin/CONTRIBUTING.md#test-plugin) to describe the testing new plugins on KB repo. Not sure if it is suitable though...